x/concentrated-liquidity/internal/math/math.go

```diff
@@ -81,20 +95,21 @@ func CalcAmount1Delta(liq, sqrtPriceA, sqrtPriceB sdk.Dec, roundUp bool) sdk.Dec
 // else
 // sqrtPriceNext = ((liquidity)) / (((liquidity) / (sqrtPriceCurrent)) + (amountRemaining))
 func GetNextSqrtPriceFromAmount0RoundingUp(sqrtPriceCurrent, liquidity, amountRemaining sdk.Dec) (sqrtPriceNext sdk.Dec) {
-	numerator := liquidity
-	product := amountRemaining.Mul(sqrtPriceCurrent)
+	if amountRemaining.Equal(sdk.ZeroDec()) {
+		return sqrtPriceCurrent
+	}
 
-	if product.Quo(amountRemaining).Equal(sqrtPriceCurrent) {
-		denominator := numerator.Add(product)
-		if denominator.GTE(numerator) {
-			numerator = numerator.Mul(sqrtPriceCurrent)
-			sqrtPriceNext = numerator.QuoRoundUp(denominator)
-			return sqrtPriceNext
+	product := amountRemaining.Mul(sqrtPriceCurrent)
+	// use precise formula if product doesn't overflow
+	// TODO: see if we even need to do this check
+	if (product.Quo(amountRemaining)).Equal(sqrtPriceCurrent) {
+		denominator := liquidity.Add(product)
+		if denominator.GTE(liquidity) {
+			return liquidity.Mul(sqrtPriceCurrent).QuoRoundUp(denominator)
 		}
 	}
-	denominator := numerator.Quo(sqrtPriceCurrent).Add(amountRemaining)
-	sqrtPriceNext = numerator.QuoRoundUp(denominator)
-	return sqrtPriceNext
+	// if the product does overflow, use less precise formula
```

Comment:
What do you mean by overflow? I think the definition is causing my confusion here

Reply:
This logic already existed prior to this PR. I was just adding comments from the uniswap code. The todo here is seeing if this less accurate formula to prevent overflows is needed in our context

Reply:
https://github.com/osmosis-labs/osmosis/issues/3887